Pyrado/pyrado/environments/mujoco/base.py

```diff
@@ -280,13 +275,15 @@ def reset(self, init_state: np.ndarray = None, domain_param: dict = None) -> np.
         if not init_state[:nq].shape == old_state.qpos.shape:  # check joint positions dimension
             raise pyrado.ShapeErr(given=init_state[:nq], expected_match=old_state.qpos)
         # Exclude everything that is appended to the state (at the end), e.g. the ball position for WAMBallInCupSim
-        if not init_state[nq : 2 * nq].shape == old_state.qvel.shape:  # check joint velocities dimension
-            raise pyrado.ShapeErr(given=init_state[nq : 2 * nq], expected_match=old_state.qvel)
+        if (
+            not init_state[nq : nq + self.init_qvel.size].shape == old_state.qvel.shape
+        ):  # check joint velocities dimension
+            raise pyrado.ShapeErr(given=init_state[nq : nq + self.init_qvel.size], expected_match=old_state.qvel)
         new_state = mujoco_py.MjSimState(
             # Exclude everything that is appended to the state (at the end), e.g. the ball position for WAMBallInCupSim
             old_state.time,
             init_state[:nq],
-            init_state[nq : 2 * nq],
+            init_state[nq : nq + self.init_qvel.size],
```

Comment:
this change should not be neccessary, please elaborate (or revert if the old one works)